app/models/reservation.rb

```diff
@@ -17,4 +17,17 @@ class Reservation < ApplicationRecord # :nodoc:
   validates :checkin_at, presence: true
   validates :checkout_at, presence: true
   validates :book_code, presence: true, uniqueness: true
+
+  # Callbacks
+  before_create :assign_available_room, if: -> { room.status == 'unavailable' }
+
+  private
+
+  def assign_available_room
+    room = Room.find_by(status: 'available', category: self.room.category)
+
+    raise StandardError, "No available room found for category #{self.room.category}" if room.nil?
+
+    self.room = room
```

Comment:
@AlbertoHdezCerezo @waywho The logic now is very simple regarding available or unavailable room on `Reservation` model. I've created some issues to handle the logic including period of reservation, housekeeping status and etc. but this logic may happen on `Room` side.

Reply:
Yeah, sounds great, this is what I meant with handling the logic to a room become available or not on Room side. I'll draw it and ping you soon! tks